[DX-955] Add use-case based SDK references across Pub/Sub pages (#3340)
ably/docs: src/pages/docs/channels/index.mdx

| Push notifications enabled | If checked, publishing messages with a push payload in the `extras` field is permitted. This triggers the delivery of a [Push Notification](/docs/push) to devices registered for push on the channel. |
| Server-side batching | If enabled, messages are grouped into batches before being sent to subscribers. [Server-side batching](/docs/messages/batch#server-side) reduces the overall message count, lowers costs, and mitigates the risk of hitting rate limits during high-throughput scenarios. |
| Message conflation | If enabled, messages are aggregated over a set period of time and evaluated against a conflation key. All but the latest message for each conflation key value will be discarded, and the resulting message, or messages, will be delivered to subscribers as a single batch once the period of time elapses. [Message conflation](/docs/messages#conflation) reduces costs in high-throughput scenarios by removing redundant and outdated messages. |

> | Message annotations, updates, deletes, and appends | If enabled, allows message [annotations](/docs/messages/annotations) to be used, as well as updates, deletes, and appends to be published to messages. Note that these features are currently in public preview. When this feature is enabled, messages will be [persisted](/docs/storage-history/storage#all-message-persistence) (necessary in order for them later be annotated or updated), and [continuous history](/docs/storage-history/history#continuous-history) features will not work.

Author: Fixed — updated to "necessary in order for them to later be annotated or updated". Also fixed a pre-existing typo on main where "for" was misspelled as "from".

> | Message annotations, updates, deletes, and appends | If enabled, allows message [annotations](/docs/messages/annotations) to be used, as well as updates, deletes, and appends to be published to messages. Note that these features are currently in public preview. When this feature is enabled, messages will be [persisted](/docs/storage-history/storage#all-message-persistence) (necessary in order for them to later be annotated or updated), and [continuous history](/docs/storage-history/history#continuous-history) features will not work.

Reviewer: You fixed the existing typo; while in the line, the parenthetical can be promoted to plain prose for easier reading:

```suggestion
| Message annotations, updates, deletes, and appends | If enabled, allows message [annotations](/docs/messages/annotations) to be used, as well as updates, deletes, and appends to be published to messages. Note that these features are currently in public preview. When this feature is enabled, messages must be [persisted](/docs/storage-history/storage#all-message-persistence) before they can be annotated or updated, so [continuous history](/docs/storage-history/history#continuous-history) features will not work.
```

Author: Applied with one tweak — used `and` instead of `so`. Persistence-required-for-annotations and continuous-history-not-working are two independent consequences of enabling the rule, not cause-and-effect, so the connector shouldn't be causal.

Author: Fixed https://github.com/ably/docs/pull/3340/commits/1a27dd4c14df6ad2f23a2ba07baaeb1b99a79004
